web/src/main/java/com/es/phoneshop/web/controller/exception/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,37 @@
+package com.es.phoneshop.web.controller.exception;
+
+import com.es.core.exception.EntityNotFoundException;
+import com.es.core.exception.OutOfStockException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<String> handleProductNotFoundException() {
+        return new ResponseEntity<>("Product not found", HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(OutOfStockException.class)
+    public ResponseEntity<String> handleOutOfStockException() {
+        return new ResponseEntity<>("Out of stock", HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(InvalidFormatException.class)
+    public ResponseEntity<String> handleInvalidFormatException() {
+        return new ResponseEntity<>("Invalid format", HttpStatus.BAD_REQUEST);
```

Comment:
Please constract user-friendly error message for future tasks, not 'invalid format' for all errors. (Note: better not to use exception handler in this case)